src/plugins/intel_cpu/src/nodes/mkldnn_reduce_node.cpp

```diff
@@ -70,6 +71,39 @@ using namespace Xbyak;
 #define GET_PTR_NCD_BASE_PTR_N_BLK const uint8_t    *in_ptr_ncd    = in_ptr_n     + src_data_size * (icb * ID + id) * IH * IW * blk_size; \
                                          uint8_t    *out_ptr_ncd   = out_ptr_n    + dst_data_size * (ocb * OD + od) * OH * OW * blk_size;
 
+namespace {
+struct ReduceKey {
```

Comment:
It's better to add input_shape & reduce_axes since they are used for [calculating dims](https://github.com/openvinotoolkit/openvino/blob/2a476f6906df36de8805ad1b1dd307b0021e73d8/src/plugins/intel_cpu/src/nodes/mkldnn_reduce_node.cpp#L1846)
```
void MKLDNNReduceNode::prepareParams() {
    src_dims = getParentEdgesAtPort(REDUCE_DATA)[0]->getMemory().getDesc().getShape().getDims();
    std::vector<int> reduce_axes;
    if (jit_mode && jit_beyond_5D) {
        reduce_axes = update_src_dims();
    } else {
        reduce_axes = raw_axes;
    }

    auto &dstMemPtr = getChildEdgeAt(0)->getMemoryPtr();
    const SizeVector &dst_dims = dstMemPtr->getDesc().getShape().getDims();
    dst_size = dstMemPtr->GetSize();
    calc_process_dst_dims(reduce_axes, dst_dims);
    if (jit_mode) {
        set_reduce_dim_flags();
    }
....

```

Reply:
Hi Yi. Since none of the params in jit_reduce_config_params is related  to input or output shapes, or we may say both jit_reduce and jit_reduce_post kernels are shape agnostic. So I think not include the shapes in the key will not lead to issues of correctness. Further discussion is always welcomed. Thanks Yi!